sharding-core/sharding-core-route/src/main/java/org/apache/shardingsphere/core/route/router/masterslave/MasterSlaveRouter.java

```diff
@@ -70,10 +70,15 @@
     }
     
     private boolean isMasterRoute(final SQLStatement sqlStatement) {
-        return containsLockSegment(sqlStatement) || !(sqlStatement instanceof SelectStatement) || MasterVisitedManager.isMasterVisited() || HintManager.isMasterRouteOnly();
+        return containsLockSegment(sqlStatement) || !(sqlStatement instanceof SelectStatement) || MasterVisitedManager.isMasterVisited()
+                || HintManager.isMasterRouteOnly() || isSelectWithoutTable(sqlStatement);
     }
 
     private boolean containsLockSegment(final SQLStatement sqlStatement) {
         return sqlStatement instanceof SelectStatement && ((SelectStatement) sqlStatement).getLock().isPresent();
     }
+
+    private boolean isSelectWithoutTable(final SQLStatement sqlStatement) {
```

Comment:
Sorry, here we need discuss again.
If it is a select SQL without table, route to slave database is fine, is it?

Reply:
If it is not a  SelectStatement, it's already route to master.   so  we don't need to make any change in Master-Slave mode?

Reply:
In this case, do we just need to consider the sharding mode?

Reply:
> If it is not a SelectStatement, it's already route to master. so we don't need to make any change in Master-Slave mode?

We have another judgement to judge select statement, right?

Reply:
Yes,  according to my understanding,  from the code :
`!(sqlStatement instanceof SelectStatement)`,
If it is a select statement, it will be routed to slave database,  no matter whether the select SQL has a table name.
For #3273, if it is fine , we can remove the `isSelectWithoutTable ` judgement.
It means we don't need to make any change with `MasterSlaveRouter`, Do I understand it correctly?

Reply:
Yes, correctly. we can do it like this way.

Reply:
OK,  thank you.